rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -269,13 +285,30 @@ def run_all_post_decoding_rules(
             transaction: EvmTransaction,
             decoded_events: list['EvmEvent'],
             all_logs: list[EvmTxReceiptLog],
+            counterparties: set[str],
     ) -> list['EvmEvent']:
         """
-        Runs all post-decoding rules from self.rules.post_decoding_rules.
-        The post-decoding rules list consists of tuples (priority, rule) and must be sorted by
-        priority in ascending order. The higher the priority number the later the rule is run.
+        Runs post-decoding rules using counterparties and the transaction.to_address as filter
+        criteria. The post-decoding rules list consists of tuples (priority, rule) and must be
+        sorted by priority in ascending order. The higher the priority number the later
+        the rule is run.
```

Comment:
Explain what the role of the counterparties is a bit more clearly.

Essentially what you say in the first sentece is what you do, but it's not easy to understand.

I would put it at the end of the docstring and say something like this. Please double check this and adjust if my understanding it not correct

```
Matches post decoding rules to all matched counterparties propagated for decoding from the decoding/enriching rules and also the counterparties associated with the transaction to address
```
Also I would remove the comment that follows the docstring, since you would be including that explanation in the docstring itself.